### PR TITLE
[DC-86] Add puppeteer test

### PR DESCRIPTION
**Pull request by tcjiang9**
foundation for TDC puppeteer test

### CODE REVIEW

File: integration-tests/tests/run-catalog-workflow.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  await signIntoTerra(page, token)
+  await dismissNotifications(page)
+
+  await click(page, clickable({ textContains: 'browse & explore' }))
+})
+
+const testCatalog = {
+  name: 'run-catalog',
+  fn: testCatalogFlowFn,
+  timeout: 15 * 60 * 1000
```
Comment (dmohs) [inline integration-tests/tests/run-catalog-workflow.js]: Is fifteen minutes necessary for this? I would expect a smaller number given what I see.

**Review by dmohs — COMMENTED** [review]
Looks good to me, and follows the existing patterns as far as I can tell. I'd like a quick check from @zarsky-broad, so I will let him approve.

**Comment by lu-c** [comment]
How do we run the puppeteer tests? Is that happening automatically in the ci? What command do I use to run it locally?

**Comment by tcjiang9** [comment]
> How do we run the puppeteer tests? Is that happening automatically in the ci? What command do I use to run it locally?

you can follow the readme here
https://github.com/DataBiosphere/terra-ui/tree/dev/integration-tests

File: integration-tests/tests/run-catalog-workflow.js (hunk 1 above)
Comment (petesantos) [reply integration-tests/tests/run-catalog-workflow.js]: I agree, I think a 1-2 minute timeout would work for now.

**Comment by lu-c** [comment]
tried running the tests, but got a "permissions denied" error, when running the vault command to extract the keys - I mean, the code looks okay, I'd rather the 2 approvals come from other ppl tho. Sorry! >.<

**Review by petesantos — APPROVED** [review]

**Comment by kyuksel** [comment]
> Looks good to me, and follows the existing patterns as far as I can tell. I'd like a quick check from @zarsky-broad, so I will let him approve.

FYI @tcjiang9 that Isaac said he's good so this PR is mergeable as far as UI code owners are concerned 😊

**Review by kyuksel — APPROVED** [review]

**Pull request merged**